src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -261,16 +287,18 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 
 	// Handles double click to edit icon change
 	// See textcell.component.ts for changing edit behavior
-	public enableActiveCellIconOnDoubleClick() {
+	public enableActiveCellEditIconOnDoubleClick() {
 		if (this.doubleClickEditEnabled) {
-			const toolbarComponent = (<CellToolbarComponent>this.cellToolbar.first);
-			const toolbarEditCellAction = toolbarComponent.getEditCellAction();
-			if (!toolbarEditCellAction.editMode) {
-				toolbarEditCellAction.editMode = !toolbarEditCellAction.editMode;
-			}
+			this.enableActiveCellEditIcon(true);
 		}
 	}
 
+	public enableActiveCellEditIcon(editMode: boolean) {
```

Comment:
Seems like `setActiveCellEditActionMode` would be more accurate since you're not always enabling it